MissionClasses.py

```diff
@@ -199,3 +220,121 @@ def __init__(self, strPropType,strMatType, nTanks, lMaxRadTank, presTank, mPropT
+        
+        # Avionics
+        mAvionics = 8*(mVehicleStart**0.361)
+        
+        # Electrical Subsystem      
+        if strLanderSize =='Small':
+            mPowerConversion = 30 # kg
+            pwrDrawLander    = 300 # W
+        else:
+            mPowerConversion = 50
+            pwrDrawLander    = 1200
+        
+        if strArrayType == 'Body':
+            pwrdenArray = 30 # w/kg
+        else: 
+            pwrdenArray = 75 # w/kg
+        
+        lTank = max(clsOxTankSet.lTankLength,clsFuelTankSet.lTankLength) # pick the maximum length of your clsOxTankSet.lTankLength and clsFuelTankSet.lTanklength
+        mWiring   = 1.058*np.sqrt(mVehicleStart)*(lTank**0.25)
+        
+        pwrTotalMargined = (1+pctMarginArray)*(pwrDrawLander+pwrDrawPayload) #the parenthesis is the sum of lander power and payload power
+        mSolarArray      = pwrTotalMargined/pwrdenArray #divide the total margined power by the density 
+        
+        nrgTotal       = pwrTotalMargined*tBattery
+        nrgTotalMargin = nrgTotal/(1-pctDepthOfDischarge)
+        mBattery       = nrgTotalMargin/nrgdenBattery
+        
+        mElectrical = mPowerConversion + mWiring + mSolarArray + mBattery
+        
+        # Propulsion
+        if strLanderSize =='Small':
+            mRCS = 20 #kg
+            mPressurization = 50 #kg
+            mFeedlines = 20 #kg
+        else:
+            mRCS = 50 #kg
+            mPressurization = 100 #kg
+            mFeedlines = 50 #kg
+            
+        if clsOxTankSet.strPropType == 'Oxygen':
+            mSOFIOx = thkSOFI*clsOxTankSet.saTotalPerTank*clsOxTankSet.nTanks*rhoSOFI
+            mMLIOx  = thkMLI*clsOxTankSet.saTotalPerTank*clsOxTankSet.nTanks*rhoMLI
+        else:
+            mSOFIOx = 0
+            mMLIOx  = thkMLI*clsOxTankSet.saTotalPerTank*clsOxTankSet.nTanks*rhoMLI
+
+        if clsFuelTankSet.strPropType == 'Hydrogen':
+            mSOFIFuel = thkSOFI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoSOFI
+            mMLIFuel  = thkMLI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoMLI
+            twEngine  = 40
+        elif clsFuelTankSet.strPropType == 'Methane':
+            mSOFIFuel = thkSOFI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoSOFI
+            mMLIFuel  = thkMLI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoMLI
+            twEngine = 50
+        elif clsFuelTankSet.strPropType == 'MMH':
+            mSOFIFuel = 0
+            mMLIFuel  = thkMLI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoMLI
+            twEngine = 50
+        elif clsFuelTankSet.strPropType == 'RP-1':
+            mSOFIFuel = 0
+            mMLIFuel  = thkMLI*clsFuelTankSet.saTotalPerTank*clsFuelTankSet.nTanks*rhoMLI
+            twEngine  = 60
+        
+        mEngine = 1/(twEngine/clsEng.thrust)/9.81
+
+        mPropulsion = mRCS + mPressurization + mFeedlines + mSOFIOx + mMLIOx + mSOFIFuel + mMLIFuel + mEngine
```

Comment:
Looks like you're missing tank mass. It should be: 

```
mPropulsion = mRCS + mPressurization + mFeedlines + mSOFIOx + mMLIOx + mSOFIFuel + mMLIFuel + mEngine \
            + clsOxTankSet.mTotal + clsFuelTankSet.mTotal 
```

Reply:
Fixed the mPropulsion assignment. Passed the unit tests. Thank you for the help!